src/library/clementine/clementinedbconnection.cpp

```diff
@@ -0,0 +1,183 @@
+#include "library/clementine/clementinedbconnection.h"
+
+#include <QFile>
+#include <QFileInfo>
+#include <QSettings>
+#include <QSqlError>
+
+#include "library/queryutil.h"
+#include "track/track.h"
+#include "util/performancetimer.h"
+
+ClementineDbConnection::ClementineDbConnection() {
+}
+
+ClementineDbConnection::~ClementineDbConnection() {
+    qDebug() << "Close Clementine database";
+    m_database.close();
+}
+
+void ClementineDbConnection::setTrackCollection(TrackCollectionManager* pTrackCollection) {
+    m_pTrackCollectionManager = pTrackCollection;
+}
+
+bool ClementineDbConnection::open(const QString& databaseFile) {
+    m_database = QSqlDatabase::addDatabase("QSQLITE", "CLEMENTINE_DB_CONNECTION");
+    m_database.setHostName("localhost");
+    m_database.setDatabaseName(databaseFile);
+    m_database.setConnectOptions("SQLITE_OPEN_READONLY");
+
+    //Open the database connection in this thread.
+    if (!m_database.open()) {
+        m_database.setConnectOptions(); // clear options
+        qWarning() << "Failed to open Clementine database." << m_database.lastError();
+        return false;
+    } else {
+        qDebug() << "Successful opened Clementine database";
+        return true;
+    }
+}
+
+QList<struct ClementineDbConnection::Playlist> ClementineDbConnection::getPlaylists() {
+    QList<struct ClementineDbConnection::Playlist> list;
+    struct ClementineDbConnection::Playlist playlist;
+
+    QSqlQuery query(m_database);
+    query.prepare("SELECT ROWID, playlists.name FROM playlists ORDER By playlists.name");
+
+    if (query.exec()) {
+        while (query.next()) {
+            playlist.playlistId = query.value(0).toString();
+            playlist.name = query.value(1).toString();
+            list.append(playlist);
+        }
+    } else {
+        LOG_FAILED_QUERY(query);
+    }
```

Comment:
This is easier to read IMHO.

```suggestion
    if (!query.exec()) {
        LOG_FAILED_QUERY(query);
        return {};
    }

    while (query.next()) {
        playlist.playlistId = query.value(0).toString();
        playlist.name = query.value(1).toString();
        list.append(playlist);
    }
```

Reply:
Btw, are you sure that `playlistId` is a string?

Reply:
yes

Reply:
added in https://github.com/mixxxdj/mixxx/pull/3173/commits/fa7de067bc68bb7dd377a7cf673b2564c880b2f3